pkg/clusteragent/autoscaling/workload/controller.go

```diff
@@ -420,7 +423,7 @@ func (c *Controller) deletePodAutoscaler(ns, name string) error {
 func (c *Controller) validateAutoscaler(podAutoscalerInternal model.PodAutoscalerInternal) error {
 	// Check that we are within the limit of 100 DatadogPodAutoscalers
 	key := podAutoscalerInternal.ID()
-	if !c.limitHeap.Exists(key) {
+	if !podAutoscalerInternal.CreationTimestamp().IsZero() && !c.limitHeap.Exists(key) {
```

Comment:
Is it possible to reach this point while have a zero `CreationTimestamp`?

Reply:
good point! i can update it to check if the value in the store is 0 if necessary but removed for now